### DISCUSSION TITLE
A Sub-Realm feature where a particular realm can have multiple sub realms under it and exchange tokens with them to give sub realm specific permissions to user

### DISCUSSION THREAD

**Discussion by 988Harsh** [keycloak/keycloak#30696] [Ideas]
Image this:

- There is a scenario which needs us to have a sso which allows login accross multiple realms.
- This is challenging to do with the current keycloak.
- But if we had a concept of sub realms, lets say I login through google idp and the first page I see the list of sub realms of which the user (me) is part of.
- On picking any sub realm I get to exchange my token (belonging to the parent realm) for the token of the sub realm.

This scenario helps with having different dashboards and roles and permissions for users while the authentication being taken care of by the same parent. Customize the DB for keycloak in such a situation and it would be a lot of fun and open up a lot of possibilities.

I believe this is like having an additional layer between groups and the realm. If there is already something like this or an extension for this in Nodejs please do let me know.

**Comment by buianhtai** [comment]
I think you can research 
- [Keycloak Organizations](https://github.com/keycloak/keycloak/issues/30180)
- [keycloak-multi-tenancy](https://github.com/anarsultanov/keycloak-multi-tenancy)
